lib/logstash/inputs/tcp.rb

```diff
@@ -136,6 +110,7 @@ def run_server(output_queue)
         # start a new thread for each connection.
         server_connection_thread(output_queue, socket)
       rescue OpenSSL::SSL::SSLError => e
+        # TODO: (colin) not sure I undestand the logic of logging & ignoring ssl errors here
```

Comment:
ok. make sense so we should probably add an explicit `socket.close rescue nil` here too then